### DISCUSSION TITLE
Moving away from Github

### DISCUSSION THREAD

**Discussion by HekkiMelody** [OCA/odoo-community.org#130] [Ideas]
Hello,

wanted to start a conversation about whether the OCA was open to considering moving away from Github.

You can find some of the reasons why you'd consider it here: [GiveUpGithub](https://giveupgithub.org/)

Being an Open-source community, I am sure many of you will be amenable to many of the arguments listed in the link above. I assume, for example, that those who kindly set up the Discord-matrix bridge did so for similar reasons. With the renewed push for Copilot all across Github, I think all these reasons have become even stronger...

I realize there is a lot of existing tooling that would need to be repurposed for a new platform such as [Codeberg](https://codeberg.org/) and that it would require a considerable amount of effort, but I though I'd at least put this out there to get a feel for how the OCA community feels about this sort of topic.

**Comment by legalsylvain** [comment]
Hi. @aleuffre. Thanks a lot for this topic. I'm totally in favor for switching from github into anotherFreeGitTool. (including hosting an dedicated OCA gitlab). (some contributors like akretion has knowledges on that topic : https://gitlab.akretion.com/)

I think : 

- It's a quite important work.
- it should be done step by step. (make CI / runboat working / work on migration script / etc...)

As a first step, we could organize a survey in the OCA (or add a question in the next survey) to know if this point is important for members.

**Comment by rvalyi** [comment]
Although I agree with the critics to Github, I give my 2 cents on this:

1. yes the amount of work would be extremely important because the OCA is much more than just a git repo. Github features, including the social ones, are used extensively... For instance when I see a new PR that would require some work but also carries some potential, I use to look at the developer or company profile, to decide if I'll invest some time helping that person to get that PR merged for instance. Are you not doing that?
3. It doesn't matter much if we get an OCA survey were people who are not going to do that work will have nice words for supporting FOSS and moving away from Github but if at the end of the day it will put yet another burden on the people doing the hard infrastructure and maintenance work or either suck away a critical part of the very limited OCA budget for this...
4. Let's have an honest look: today most of the Github critics you can read in https://sfconservancy.org/GiveUpGitHub also apply to the Odoo SA company. IMHO that would be hypocrite if most of the OCA was built atop of Odoo CE built and driven by Odoo SA and we would focus vital resources to emancipate from Github instead which is just a very minor part of the "not FOSS friendly problem".
5. There is no free lunch. If we move away from the "you are the product" Github platform, we should be sure it will be balanced by enough resources like donations and that things will be optimized enough as they can be with massive platforms like Github. Again if the OCA was just a small git repo with 10 active contributors like some GNU tool, well things would be different and any git tool would cut it...

At the end, I would say we should be aware when building our tooling so we don't increase the Github dependency too much. I think people are aware of this, Now as for moving away, I think we have tons of other priorities sadly...
